services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowDiversitySpec.groovy

```diff
@@ -0,0 +1,417 @@
+        def flow1 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow1)
+        def flow1Path = PathHelper.convert(northbound.getFlowPath(flow1.id))
+
+        and: "Make all min alternative paths less preferable by 1 for diverse flows"
+        def altPaths = database.getPaths(srcSwitch.dpId, dstSwitch.dpId)*.path
+        altPaths.remove(flow1Path)
+        def flow1PathCost = pathHelper.getCost(flow1Path) + diversityIslWeight + diversitySwitchWeight * 2
+        def minAltPathCost = pathHelper.getCost(altPaths.min { it.size() }) + diversitySwitchWeight * 2
+        if (flow1PathCost > minAltPathCost) {
+            int minAltPathNodeCount = altPaths.min { it.size() }.size()
+            int difference = flow1PathCost - minAltPathCost
+            altPaths.findAll { it.size() == minAltPathNodeCount }.each {
+                def isl = pathHelper.getInvolvedIsls(it)[0]
+                int islCost = database.getIslCost(isl)
+                northbound.updateLinkProps([new LinkPropsDto(isl.srcSwitch.dpId.toString(), isl.srcPort,
+                        isl.dstSwitch.dpId.toString(), isl.dstPort, ["cost": (islCost + difference + 1).toString()])])
+            }
+        }
+
+        when: "Create the second flow with diversity enabled"
+        def flow2 = flowHelper.randomFlow(srcSwitch, dstSwitch, false, [flow1]).tap { it.diverseFlowId = flow1.id }
+        flowHelper.addFlow(flow2)
+        def flow2Path = PathHelper.convert(northbound.getFlowPath(flow2.id))
+
+        then: "The flow is built through the most preferable path (path of the first flow)"
+        flow2Path == flow1Path
+
+        when: "Create the third flow with diversity enabled"
+        if (flow1PathCost < minAltPathCost) {
+            int difference = minAltPathCost - flow1PathCost
+            def isl = pathHelper.getInvolvedIsls(flow1Path)[0]
+            northbound.updateLinkProps([new LinkPropsDto(isl.srcSwitch.dpId.toString(), isl.srcPort,
+                    isl.dstSwitch.dpId.toString(), isl.dstPort, ["cost": (flow1PathCost + difference + 1).toString()])])
+        }
+        def flow3 = flowHelper.randomFlow(srcSwitch, dstSwitch, false, [flow1, flow2]).tap {
+            it.diverseFlowId = flow2.id
+        }
+        flowHelper.addFlow(flow3)
+        def flow3Path = PathHelper.convert(northbound.getFlowPath(flow3.id))
+
+        then: "The flow is built through one of min alternative paths because they are preferable already"
+        def involvedIsls = [flow2Path, flow3Path].collectMany { pathHelper.getInvolvedIsls(it) }
+        flow3Path != flow2Path
+        involvedIsls.unique(false) == involvedIsls
+
+        and: "Delete flows and link props"
+        [flow1, flow2, flow3].each { flowHelper.deleteFlow(it.id) }
+        northbound.deleteLinkProps(northbound.getAllLinkProps())
+    }
+
+    def "Able to get flow paths with correct overlapping segments stats (casual flows)"() {
```

Comment:
This overlaps with the first test, isn't it? Moreover, this test will get more complete if you explicitly show the `then: "All flows have different paths"` verification from the first test (which is the only difference between the tests). Then it looks more obvious _why_ the overlapping segments have given values

Reply:
Ok, just saw your comment. We can ofc leave as is, this is not crucial